server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CCSRemoteClusterClientRoleIT.java

```diff
@@ -0,0 +1,89 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.ccs;
+
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodeRole;
+import org.elasticsearch.index.query.MatchAllQueryBuilder;
+import org.elasticsearch.test.AbstractMultiClustersTestCase;
+import org.elasticsearch.test.InternalTestCluster;
+import org.elasticsearch.test.NodeRoles;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
+import static org.hamcrest.Matchers.equalTo;
+
+public class CCSRemoteClusterClientRoleIT extends AbstractMultiClustersTestCase {
+
+    @Override
+    protected Collection<String> remoteClusterAlias() {
+        return List.of("cluster_a");
+    }
+
+    private int indexDocs(Client client, String index) {
+        assertAcked(client.admin().indices().prepareCreate(index));
+        int numDocs = between(1, 10);
+        for (int i = 0; i < numDocs; i++) {
+            client.prepareIndex(index).setSource("f", "v").get();
+        }
+        client.admin().indices().prepareRefresh(index).get();
+        return numDocs;
+    }
+
+    public void testRemoteClusterClientRole() throws Exception {
+        final int demoDocs = indexDocs(client(LOCAL_CLUSTER), "demo");
+        final int prodDocs = indexDocs(client("cluster_a"), "prod");
+        final InternalTestCluster localCluster = cluster(LOCAL_CLUSTER);
+        if (randomBoolean()) {
+            localCluster.startDataOnlyNode();
+        }
+        final String nodeWithoutRemoteClusterClientRole = localCluster.startNode(NodeRoles.onlyRole(DiscoveryNodeRole.DATA_ROLE));
+        final IllegalStateException error = expectThrows(IllegalStateException.class, () ->
+            localCluster.client(nodeWithoutRemoteClusterClientRole)
+                .prepareSearch("demo", "cluster_a:prod")
+                .setQuery(new MatchAllQueryBuilder())
+                .setAllowPartialSearchResults(false)
+                .setSize(1000)
+                .get());
+        assertThat(error.getMessage(),
+            equalTo("node [" + nodeWithoutRemoteClusterClientRole + "] does not have the remote cluster client role enabled"));
```

Comment:
would it be possible to verify what status code would get returned to users at the REST layer?

Reply:
I've strengthened the assertion in https://github.com/elastic/elasticsearch/pull/60351/commits/78ba655192c7788b0082da15526f0ba0bcc79691. I don't know how to verify the rest status in this IT test. Any suggestion?